.mvn/wrapper/MavenWrapperDownloader.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2007-present the original author or authors.
```

Comment:
I'm guessing this class came from somewhere else, would be good to have a link in here somewhere with a pointer to where exactly. "original author" doesn't really mean anything in this context.

Reply:
True, but since this is autogenerated code I wouldn't change this (?)

Reply:
Oh right, I didn't realise it was autogenerated. Does it need to be in version control then? I guess it can't be autogenerated only when needed?

Reply:
Usually folks commit this file to open-source projects which are using the Maven Wrapper.
Otherwise one needs to run the following locally to generate the `.mvn` dir:
```
mvn -N io.takari:maven:wrapper
```

It's more for facilitating end users and contributors.

Reply:
OK, I've never used the Maven wrapper, if this is how it's usually done in other projects too then it's fine as is.